src/services/shop/shop-router.test.ts

```diff
@@ -0,0 +1,431 @@
+
+        // Verify all items were updated correctly
+        const updatedItem1 = await Models.ShopItem.findOne({ itemId: TESTER_SHOP_ITEM.itemId });
+        const updatedItem2 = await Models.ShopItem.findOne({ itemId: secondItem.itemId });
+        expect(updatedItem1?.quantity).toBe(TESTER_SHOP_ITEM.quantity - 1);
+        expect(updatedItem2?.quantity).toBe(TESTER_SHOP_ITEM.quantity - 2);
+
+        // Verify total points deduction (TESTER_SHOP_ITEM.price * 1 + secondItem.price * 2)
+        const updatedProfile = await Models.AttendeeProfile.findOne({ userId: TESTER_PROFILE.userId });
+        const expectedPoints = TESTER_PROFILE.points - (TESTER_SHOP_ITEM.price + secondItem.price * 2);
+        expect(updatedProfile?.points).toBe(expectedPoints);
+    });
+});
+
+describe("POST /shop/cart/:itemId", () => {
+    it("allows user to add new item to cart", async () => {
+        const response = await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject({
+            ...TESTER_SHOP_ORDER,
+            quantity: [3],
+        });
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(updatedOrder?.quantity[0]).toBe(3);
+    });
+
+    it("increases quantity when adding existing item to cart", async () => {
+        // First addition
+        await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        // Second addition
+        const response = await postAsAttendee("/shop/cart/test-item-1").expect(StatusCode.SuccessOK);
+
+        expect(JSON.parse(response.text)).toMatchObject({
+            ...TESTER_SHOP_ORDER,
+            quantity: [4],
+        });
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId: TESTER_PROFILE.userId });
+        expect(updatedOrder?.items).toContain(TESTER_SHOP_ITEM.itemId);
+        expect(updatedOrder?.quantity[0]).toBe(4);
+    });
+
+    it("returns NotFound for non-existent item", async () => {
+        await postAsAttendee("/shop/cart/non-existent-item").expect(StatusCode.ClientErrorNotFound);
+    });
+
+    it("returns BadRequest when insufficient shop quantity", async () => {
+        await Models.ShopItem.create({
+            ...TESTER_SHOP_ITEM,
+            itemId: "out-of-stock-item",
+            quantity: 0,
+        });
+
+        await postAsAttendee("/shop/cart/out-of-stock-item").expect(StatusCode.ClientErrorBadRequest);
+    });
+
+    it("returns PaymentRequired when insufficient points", async () => {
+        await Models.ShopItem.create({
+            ...TESTER_SHOP_ITEM,
+            itemId: "expensive-item",
+            price: 2000,
+            quantity: 1,
+        });
+
+        await postAsAttendee("/shop/cart/expensive-item").expect(StatusCode.ClientErrorPaymentRequired);
```

Comment:
you shouldn't be checking just status code - also check the error returned. See other tests for reference.